contracts/access/manager/AccessManager.sol

```diff
@@ -0,0 +1,339 @@
+     */
+    function updateGroupName(uint8 group, string memory name) public virtual onlyDefaultAdmin {
+        require(group != _GROUP_PUBLIC, "AccessManager: built-in group");
+        require(hasGroup(group), "AccessManager: unknown group");
+        emit GroupUpdated(group, name);
+    }
+
+    /**
+     * @dev Returns true if the group has already been created via {createGroup}.
+     */
+    function hasGroup(uint8 group) public view virtual returns (bool) {
+        return _getGroup(_createdGroups, group);
+    }
+
+    /**
+     * @dev Returns a bitmap of the groups the user has. See note on bitmaps above.
+     */
+    function getUserGroups(address user) public view virtual returns (bytes32) {
+        return _userGroups[user] | _groupMask(_GROUP_PUBLIC);
+    }
+
+    /**
+     * @dev Grants a user a group.
+     *
+     * Emits {RoleGranted} with the role id of the group, if wasn't already held by the user.
+     */
+    function grantGroup(address user, uint8 group) public virtual {
+        grantRole(_encodeGroupRole(group), user); // will check msg.sender
+    }
+
+    /**
+     * @dev Removes a group from a user.
+     *
+     * Emits {RoleRevoked} with the role id of the group, if previously held by the user.
+     */
+    function revokeGroup(address user, uint8 group) public virtual {
+        revokeRole(_encodeGroupRole(group), user); // will check msg.sender
+    }
+
+    /**
+     * @dev Allows a user to renounce a group.
+     *
+     * Emits {RoleRevoked} with the role id of the group, if previously held by the user.
+     */
+    function renounceGroup(address user, uint8 group) public virtual {
+        renounceRole(_encodeGroupRole(group), user); // will check msg.sender
+    }
+
+    /**
+     * @dev Returns a bitmap of the groups that are allowed to call a function of a target contract. If the target
+     * contract is in open or closed mode it will be reflected in the return value.
+     */
+    function getFunctionAllowedGroups(address target, bytes4 selector) public view virtual returns (bytes32) {
+        RestrictedMode mode = getContractMode(target);
+        if (mode == RestrictedMode.Open) {
+            return _groupMask(_GROUP_PUBLIC);
+        } else if (mode == RestrictedMode.Closed) {
+            return 0;
+        } else {
+            return _allowedGroups[target][selector];
+        }
+    }
+
+    /**
+     * @dev Changes whether a group is allowed to call a function of a contract, according to the `allowed` argument.
+     * The caller must be the default admin.
+     */
+    function setFunctionAllowedGroup(
+        address target,
+        bytes4[] calldata selectors,
+        uint8 group,
+        bool allowed
+    ) public virtual onlyDefaultAdmin {
+        require(_contractMode[target] == RestrictedMode.Custom, "AccessManager: target in special mode");
+        for (uint256 i = 0; i < selectors.length; i++) {
```

Comment:
We may want to restrict selectors length by some `MAX_SELECTORS`, similar to `_maxBatchSize`in `ERC721Consecutive` if either:

1. It may cause issues to emit multiple `GroupAllowed` at once.
2. This function can be DoS

I would discard number 2, but at least we should add a not for other contracts relying on this function.